apps/google-analytics-4/frontend/src/components/config-screen/api-access/display/DisplayServiceAccountCard.tsx

```diff
@@ -0,0 +1,255 @@
+  const [unspecifiedDataError, setUnspecifiedDataError] = useState<GAErrorApiType>();
+
+  // NOTE: Due to a bug installation parameters are not available at sdk.parameters.installation form the config screen
+  // location. Therefore we must pass down the values directly to the useApi hook. If the bug is fixed this won't be
+  // necessary
+  const api = useApi(serviceAccountKeyId, serviceAccountKey);
+
+  const verifyAdminApi = useCallback(async () => {
+    try {
+      setIsLoading(true);
+      const fetchedAccountSummaries = await api.listAccountSummaries();
+      if (fetchedAccountSummaries.res && !fetchedAccountSummaries.res.ok) {
+        setAdminApiError(fetchedAccountSummaries.errors);
+        return;
+      }
+
+      onAccountSummariesChange(fetchedAccountSummaries);
+      setAdminApiError(undefined);
+      setUnspecifiedAdminError(undefined)
+    } catch (e: any) {
+      setUnspecifiedAdminError({
+        details: null,
+        errorType: GoogleErrorApiErrorTypes.AdminAPI,
+        message: 'Unknown Google Admin API Error',
+      });
+    } finally {
+      setIsLoading(false);
+    }
+
+    return () => {
+      setAdminApiError(undefined);
+      setUnspecifiedAdminError(undefined)
+      setIsLoading(false);
+    }
+  
+  // It wants to add onAccountSummariesChange as a dependency but this will cause an infinite re-render
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [api]);
+
+  const verifyDataApi = useCallback(async () => {
+    try {
+      setIsLoading(true);
+      const pagedData = await api.getPageData();
+
+      // If we get a 400 it means auth check passes and api has been enabled
+      if (pagedData.res && !pagedData.res.ok && pagedData.res.status !== 400) {
+        setDataApiError(pagedData.errors);
+        return;
+      }
+
+      setDataApiError(undefined);
+      setUnspecifiedDataError(undefined)
+    } catch (e: any) {
+      setUnspecifiedDataError({
+        details: null,
+        errorType: GoogleErrorApiErrorTypes.AdminAPI,
+        message: 'Unknown Google Admin API Error',
+      });
+    } finally {
+      setIsLoading(false);
+    }
+
+
+    return () => {
+      setDataApiError(undefined);
+      setUnspecifiedDataError(undefined)
+      setIsLoading(false);
+    }
+  }, [api]);
+
+  useEffect(() => {
+    verifyAdminApi();
+    verifyDataApi();
+  }, [verifyAdminApi, verifyDataApi]);
+
+  const handleApiTestClick = () => {
+    verifyAdminApi();
+    verifyDataApi();
+  };
+
+  interface BadgeNoteType {
+    badgeLabel: string;
+    noteMessage: string;
+  }
+
+  const RenderSimpleBadgeNote = ({ badgeLabel, noteMessage }: BadgeNoteType) => {
+    return (
+      <Stack spacing="spacingL" marginBottom="none" alignItems="flex-start" flexDirection="column">
+        <Badge variant="negative">{badgeLabel}</Badge>
+        <Note variant="warning">{noteMessage}</Note>
+      </Stack>
+    );
+  };
+
+  const RenderStatusInfo = () => {
```

Comment:
I think we can discuss the design of this. We can leave as is for this iteration if we want to though, and just drop a TODO comment and a sub task in our backlog to refine the UX design of the error badges etc. (To simplify discussion on this PR and focus more on the underlying error handling mechanics)

Reply:
Added TODO on line 141